### ISSUE TITLE
Code block cannot specify type without displaying folder path string

### ISSUE THREAD

**Issue by rebekahlow-jy** [MarkBind/markbind#72]
**Issue:**
Code blocks cannot specify type, as shown in the picture below - 
![type](https://user-images.githubusercontent.com/13508274/27018588-4bfeedee-4f64-11e7-9105-875eae860b3c.png)

**Temporary Fix:**
The temporary fix used for this is to add a space between the type declaration and the `<` and `>` signs, as shown in the picture below - 
![fix](https://user-images.githubusercontent.com/13508274/27018564-2cca6f84-4f64-11e7-847a-9ac7e45c06fa.png)

**Comment by Gisonrg** [comment]
This is due to the parsing mechanism. We parsed the whole file as DOM first in order to access all the tags and markups. I guess the brackets must be considered as an DOM element as well from the view of the parser, which results the wrong parsing...
I don't think this is a trivial fix (unfortunately), so I suggest using the temporary fix for now.

**Comment by damithc** [comment]
Does the problem happen to html code segments such as this too?
```html
<span>abc</span>
```

**Comment by acjh** [comment]
> Does the problem happen to html code segments such as this too?
> `<span>abc</span>`

Yes.

**Comment by Gisonrg** [comment]
This was an issue introduced by the attempt to resolving containing MarkBind site. Extra attributes are added to the code block which caused them to be rendered wrongly. 
Let me think about how to fix this...

**Comment by damithc** [comment]
Oops, then it is a major problem. Luckily we don't have lot of code samples of that nature. Not sure how to solve it though. May be use some other means (i.e. instead of ` ``` `) for formatting code? The current code formatter doesn't seem to do a good job anyway (highlights some random words). like the one we used in the previous cs2103 website? It's also nice if the code formatting supported line numbering, highlighting etc.

**Comment by Gisonrg** [comment]
I have figured it out. It should be able to fixed by changing the parsing order:
1. include/parse
2. resolve baseUrl (remove all extra attributes added, which clean up the dirty code block)
3. render

Let me give it a try, it should work.

For the code highlighting issue, we are using a popular library named `highlight.js`, which they claimed missing line number is a [feature](http://highlightjs.readthedocs.io/en/latest/line-numbers.html)...

We could specify the code language for highlighting using syntax like:

<pre>
```Java
class Payroll1 {
    ArrayList<Admin> admins;
    ArrayList<Academic> academics;
    // ...

    void adjustSalary(int byPercent) {
        for (Admin ad: admins) {
            ad.adjustSalary(byPercent);
        }
        for (Academic ac: academics) {
            ac.adjustSalary(byPercent);
        }
    }
}
```
</pre>
It might help to correct the highlighting.

Let me see if there is any better libraries to use :)

**Comment by damithc** [comment]
> I have figured it out.

Nice 💯

**Comment by damithc** [comment]
> We could specify the code language for highlighting using syntax like

Java is no problem. But other things such as plain text highlight random words.
![image](https://user-images.githubusercontent.com/1673303/28238842-5351eb38-698f-11e7-9f85-3932ad64966d.png)

May be there is a way to turn it off that we are unaware of. I remember trying ` ```txt` but it didn't work.

**Comment by damithc** [comment]
Anyway, we can keep using the current library `highlight.js` until we find a better one. Not urgent. The current one does a decent job.

**Comment by Gisonrg** [comment]
> But other things such as plain text highlight random words.

How about turning off the auto language detection, which makes everything explicit (i.e. we must specify the language name in the code block).

**Comment by damithc** [comment]
> How about turning off the auto language detection, which makes everything explicit (i.e. we must specify the language name in the code block).

yes, that might work.

**Comment by Gisonrg** [comment]
Fixed in 1.3.3. Let me know if it happen again :)
